examples/platforms/da15000/platform.c

```diff
@@ -143,6 +143,7 @@ bool PlatformPseudoResetWasRequested(void)
     return false;
 }
 
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
```

Comment:
How about feature flag this on the commissioner vs. joiner roles?

Reply:
I actually removed notifier from radio only mode, because there's no thread state at all.
Besides, I'm wondering if we should divide platform code into two layers, app layer and driver layer. app layer should only call openthread upper layer API and driver layer should not.

Reply:
The point of this specific comment is that this code is not needed even in the MTD/FTD configurations if DTLS is not being used.

All that said, this is example code, so I'm okay to leave it as you have it.  As you noted, the platform code could be better designed.  The platform code here is intended to be just an example.